--- a/Makefile.am
+++ b/Makefile.am
@@ -16,6 +16,7 @@
 
 DESTDIR =
 LIBEXECDIR = @LIBEXECDIR@
+SYSCONFDIR = @SYSCONFDIR@
 MYSQL_CFLAGS = @MYSQL_CFLAGS@
 MYSQL_LIBS = @MYSQL_LIBS@
 MYSQL_LIB_DIR = @MYSQL_LIB_DIR@
@@ -33,10 +34,13 @@
 $(DESTDIR)$(LIBEXECDIR) :
 	mkdir -p $@
 
-install : all $(DESTDIR)$(LIBEXECDIR)
-	$(APXS) -i mod_osa.la
-	cp osa.load  /etc/apache2/mods-available/.
-	cp osa.conf  /etc/apache2/mods-available/.
+$(DESTDIR)$(SYSCONFDIR)/mods-available/ :
+	mkdir -p $@
+
+install : all $(DESTDIR)$(LIBEXECDIR) $(DESTDIR)$(SYSCONFDIR)/mods-available/
+	$(APXS) -i -S LIBEXECDIR=$(DESTDIR)$(LIBEXECDIR) mod_osa.la
+	cp osa.load  $(DESTDIR)$(SYSCONFDIR)/mods-available/.
+	cp osa.conf  $(DESTDIR)$(SYSCONFDIR)/mods-available/.
 
 clean :
 	-rm -rf .libs
--- a/configure.ac
+++ b/configure.ac
@@ -51,6 +51,8 @@
 AC_SUBST(LDLIBS)
 
 LIBEXECDIR="`${APXS} -q LIBEXECDIR`"
+SYSCONFDIR="`${APXS} -q SYSCONFDIR`"
 AC_SUBST(LIBEXECDIR)
+AC_SUBST(SYSCONFDIR)
 AC_OUTPUT(Makefile)
 
